### PR TITLE
`Team exercises`: Add csv file team import and provide better error messages

### LINKED ISSUE #4255: Team import: Support CSV files
### Is your feature request related to a problem?

Artemis does not support team formation by students, thus Moodle is often used for this purpose. Unfortunately, Moodle generates CSV files, which are not supported by Artemis.

### Describe the solution you'd like

Implement team import from CSV files.

### Describe alternatives you've considered

_No response_

### Additional context

This feature was already requested in #1994

### LINKED ISSUE #4256: Team import: Better error handling
### Is your feature request related to a problem?

The current error handling during team import is insufficient:
1. If a value in the input file is incorrect, only a generic error message is shown, which does not specify the incorrect value. This makes debugging difficult.
2. If the input file contains students which are not currently enrolled in the course, the import fails.

### Describe the solution you'd like

1. Show more detailed error messages.
2. Allow to skip students, which are not enrolled in the course or offer to enroll them automatically.

### Describe alternatives you've considered

_No response_

### Additional context

These problems arise if files from external tools, like Moodle, are used.

### PR DESCRIPTION
**Pull request by Hialus**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Closes #4255 
Closes #4256 
In order to use external tools to create teams of students for an exercise, Artemis has an import teams function. This function however only supports json files. Tools like Moodle however export their teams as csv files. So it would be useful to be able to import these files directly.
Also, the error messages when the imported file was malformed were quite poor. Since these files can contains hundreds or thousands of entries it is important to at least tell the user which entry is responsible for the error.

### Description

For the import of csv files I simply changed the file import to also accept csv files which are then parsed with PapaParse. They accept the same keys as the json file requires, but also accept some additional keys like `Matrikelnummer` instead of `registrationNumber`. For example this will allow instructors to directly import files from Moodle (The format of which was provided in #1994)
For the error messages I changed the language keys to also include parameters that describe which entry is responsible for the error. The error messages now look similar to this:
> The team name 1Team XYZ in entry 101 does not start with a letter.

### Steps for Testing

Prerequisites:
- 1 Instructor
- A course with a couple of registered students
- 1 Exercise with teams enabled
- Some valid and invalid json files (You can create these by exporting teams from another exercise)
- Some valid and invalid csv files (You can create these by exporting teams from another exercise and convert the json file here: https://www.convertcsv.com/json-to-csv.htm)

1. Log in to Artemis
2. Navigate to the course management exercise page of the exercise that has teams enabled
3. Navigate to the teams page of this exercise
4. Click on Import Teams
5. Import the csv or json files
    - The valid files should import with no problem
    - The invalid files should provide proper error messages

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

File: Lines % | Branches %
teams-import-from-file-form.component.ts:  95.95% | 73.33%

### Screenshots

No direct UI changes. Only the error messages are different.

### CODE REVIEW

**Review by mallgaier — APPROVED** [review]
Tested together with @lBruchmann and @Strohgelaender.  Import tested using JSON. Users with incorrect usernames are ignored and not imported.
If a team name does not start with a letter, a corresponding browser error message is displayed.

**Review by lBruchmann — DISMISSED** [review]
Tested together with @Strohgelaender and @mallgaier. Tested on ts1. I followed the test steps. Importing valid and invalid (both .json and .csv files) works as expected. We have one small comment, see @Strohgelaender .

**Review by Strohgelaender — DISMISSED** [review]
Tested in testing session with @lBruchmann @mallgaier on ts1. The import worked for both csv and json files and the enhanced error messaged were shown if invalid input was provided. But directly pasting the example provided in #1994 did not work, since it uses both `;` and `,` as a separator. We might have to recheck whether or not this is indeed the format that is provided by moodle, if yes the import stil won't work.

**Comment by Hialus** [comment]
> Tested in testing session with @lBruchmann @mallgaier on ts1. The import worked for both csv and json files and the enhanced error messaged were shown if invalid input was provided. But directly pasting the example provided in #1994 did not work, since it uses both `;` and `,` as a separator. We might have to recheck whether or not this is indeed the format that is provided by moodle, if yes the import stil won't work.

If this is actually the format that Moodle uses the user will have to fix this before importing it.
While the tool used for parsing the CSV file (Papa Parse) tries to automatically detect the used separator, it does not support different separators in one file.

**Review by Garfield96 — CHANGES_REQUESTED** [review]
Nice PR. Thanks for fixing my issues. Please update the UI and documentation to show the user which input formats are supported now.

**Comment by Hialus** [comment]
I have removed the code duplication in the test cases and added the supported file types to the import dialog.
However, I could not find any documentation about the teams import feature, so I cannot update it. And it would be out of scope for this PR to add an entire documentation page about the teams feature.

File: src/test/javascript/spec/component/team/teams-import-from-file-form.component.spec.ts [hunk 1]
```diff
@@ -49,81 +44,94 @@ describe('TeamsImportFromFileFormComponent', () => {
...
     describe('onFileLoadImport', () => {
-        let convertTeamsStub: SinonStub;
+        let convertTeamsStub: jest.SpyInstance;
         let teams: Team[];
         let reader: FileReader;
-        let getElementStub: SinonStub;
+        let getElementStub: jest.SpyInstance;
         const element = document.createElement('input');
-        const control = { ...element, value: 'test' };
+        let control: any;
```
Comment (julian-christl) [inline src/test/javascript/spec/component/team/teams-import-from-file-form.component.spec.ts:71]: Please avoid `any`.

File: src/main/webapp/i18n/de/team.json [hunk 2]
```diff
@@ -41,8 +41,8 @@
                 "minlength": "Der Name muss mindestens 3 Zeichen lang sein."
             },
             "teamName": {
-                "missingTeamName": "Alle Studierenden müssen einem Teamnamen zuordnet sein.",
-                "pattern": "Der Teamname muss mit einem Buchstaben beginnen."
+                "missingTeamName": "Studierende:r {{ studentName }} in Eintrag {{ entryNr }} ist kein Teamname zuordnet.",
```
Comment (Malyuk-A) [inline src/main/webapp/i18n/de/team.json:44]:
```suggestion
                "missingTeamName": "Studierende:r {{ studentName }} in Eintrag {{ entryNr }} ist kein Teamname zugeordnet.",
```

**Review by Malyuk-A — COMMENTED** [review]
small type, else approve

File: src/main/webapp/app/exercises/shared/team/teams-import-dialog/teams-import-from-file-form.component.ts [hunk 3]
```diff
@@ -73,11 +104,58 @@ export class TeamsImportFromFileFormComponent {
...
+        );
+    }
+
     /**
      * Convert imported team list to normal teams
      */
     convertTeams(importTeam: StudentWithTeam[]): Team[] {
         const teams: Team[] = [];
+        let entryNr = 1;
         importTeam.forEach((student) => {
```
Comment (julian-christl) [inline src/main/webapp/app/exercises/shared/team/teams-import-dialog/teams-import-from-file-form.component.ts:159]:
```suggestion
        importTeam.forEach((student, index) => {
```
You can the remove `entryNr` and use `index + 1`

**Review by julian-christl — APPROVED** [review]
Code looks good now

**Review by Malyuk-A — APPROVED** [review]
Code looks good

**Review by daniels98it — APPROVED** [review]
No complaint on code 👍

**Review by Garfield96 — APPROVED** [review]

**Pull request merged**